src/Compiler/CompilerOptions.cpp

```diff
@@ -1025,6 +1025,20 @@ std::string getToolPath(
   return toolPathMap.at(tool);
 }
 
+std::string getToolLibraryPath(
+    const std::string &tool, bool flag /*false by default*/) {
+  std::string toolPath = getToolPath(tool, flag);
+  // Strip "tool-name".
+  llvm::StringRef toolPathRef(toolPath);
+  llvm::SmallVector<char> path(toolPathRef.begin(), toolPathRef.end());
+  llvm::sys::path::remove_filename(path);
+  // Add "../lib".
+  llvm::sys::path::append(path, "../lib");
+  // "Remove bin/..".
+  llvm::sys::path::remove_dots(path, /*remove dot dot*/ true);
+  return std::string(path.data(), path.size());
+}
+
```

Comment:
@gongsu832, thanks for your detailed review.

- `getLibraryPath()` gets you `.../onnx-mlir/(Debug|Release)/lib`
- `getToolLibraryPath()` gets you `.../llvm-project/build/lib`

I need the second one as this is where the OpenMP runtime resides. Would you like me to better comment the two?

Reply:
In that case, `getToolLibraryPath` may not be the best name since `Tool` here is not just llvm tools like `llc` but also other tools like `ar`, `cxx`, `jar`, etc. Since onnx-mlir depends on llvm, an llvm lib is logically an internal matter for onnx-mlir. So perhaps a better alternative is to extend the `getLibraryPath` function to take an optional argument that, when set, would go through the new code to find llvm lib path. Otherwise, it finds the onnx-mlir lib path as it does today.